src/Umbraco.Web/PublishedCache/ContentCacheContentRouter.cs

```diff
@@ -139,7 +134,7 @@ private string GetRouteByIdInternal(IPublishedCache2 publishedCache, IDomainCach
             }
 
             // at this point this will be the urlSegment of the root, no segment indicates this is not published when this is a variant
-            if (urlSegment.IsNullOrWhiteSpace()) return null;
+            if (urlSegment.IsNullOrWhiteSpace()) return new RouteByIdResult(RoutingOutcome.NotFound);
```

Comment:
This could be a static instead of a new struct like return `RouteByIdResult.NotFound;`